torch_xla/csrc/tensor.cpp

```diff
@@ -543,9 +583,28 @@ std::shared_ptr<XLATensor> XLATensor::t() const {
 
 std::shared_ptr<XLATensor> XLATensor::view(
     tensorflow::gtl::ArraySlice<const xla::int64> output_size) const {
-  return Create(std::make_shared<ir::ops::View>(ir::NodeOperand(GetIrNode()),
-                                                output_size),
-                GetDevice());
+  if (data_->view != nullptr) {
+    // Handle view of a view. This node is already a view, so use the view alias
+    // to create the new IR Node.
+    std::vector<xla::int64> complete_dimensions =
+        GetCompleteShape(output_size, data_->view->shape.dimensions());
+    xla::Shape shape = xla::ShapeUtil::MakeShape(
+        data_->view->shape.element_type(), complete_dimensions);
+    return Create(std::make_shared<View>(std::move(shape), data_->view->alias),
+                  GetDevice());
+  }
+  // This node is not a view, and creating a view forks the current node into
+  // becoming one itself. This means creating an alias with the current IR Node,
+  // and using the same alias for the created IR Node.
+  ir::NodePtr ir_node = GetIrNode();
```

Comment:
I'd move the code starting here to a separate method, it's pretty subtle and we'll probably have more aliasing operations anyway.

Reply:
Can' split this function as both parts (up/low) go hand in hand.
I agree that this could be refactored, but until we have a new aliasing concrete case/op, it's hard to find the right API boundary.